### PR TITLE
feat: correct rule 154 (#505)

### LINKED ISSUE #505: Rule #154 is incorrect for headers
Rule [#154 - Should: Define Collection Format of Query Parameters and Headers](https://opensource.zalando.com/restful-api-guidelines/#154) is incorrect for headers. [Open API 3.0](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#style-values) does not support `style: form` for headers. The equivalent specification to query parameters `style: form` for headers is `style: simple`.

In addition, I like to challenge whether we should enforce collection formats by making this a `MUST` rule instead of `SHOULD`.

### PR DESCRIPTION
**Pull request by tkrop**
fixes #505.

In addition, this pull request suggests that users explicitly *must* define the collection format. The OpenAPI specification 2.0 was dropped, to foster migration to 3.0.x after supporting it now for nearly a year as default.

### CODE REVIEW

File: chapters/http-requests.adoc [hunk 1]
```diff
@@ -331,19 +331,36 @@ original one implementing a well defined behavior. Using {204} without content
...
+either by providing a comma-separated list of values (`csv`) or by repeating
+the parameter multiple times with different values (`multi`) as follows:
+
+[,cols="14%,30%,39%,17%",options="header",]
+|=========================================================================
+| Parameter Type | Comma-separated Values | Multiple Parameters | Standard
+| Header | `Header: value1,value2` | `Header: value1, Header: value2`
+| {RFC-7230}#section-3.2.2[RFC 7230 Section 3.2.2]
+
+| Path | `key=value1,value2` | `/key=value1;key=value2`
```
Comment (maxim-tschumak) [inline chapters/http-requests.adoc]: The rule title don't say anything about path parameters - we should either add path params to the title or remove those from the table

File: chapters/http-requests.adoc [hunk 2]
```diff
@@ -331,19 +331,36 @@ original one implementing a well defined behavior. Using {204} without content
...
-[,cols="20%,20%,25%,35%",options="header",]
-|=======================================================================
-|Description |OpenAPI 3.0 |OpenAPI 2.0 |Example
-|Comma separated values |`style: form, explode: false` |`collectionFormat: csv` |`?param=value1,value2`
-|Multiple parameters |`style: form, explode: true` |`collectionFormat: multi` |`?param=value1&param=value2`
-|=======================================================================
+== {MUST} Define Collection Format of Query Parameters and Headers
+
+Header, path, and query parameters  allow to provide a collection of values,
+either by providing a comma-separated list of values (`csv`) or by repeating
```
Comment (maxim-tschumak) [inline chapters/http-requests.adoc]: `csv`, `multi` - this is Swagger (Open API 2) terminology - I would just remove it.

File: chapters/http-requests.adoc [hunk 3]
```diff
@@ -331,19 +331,36 @@ original one implementing a well defined behavior. Using {204} without content
...
+either by providing a comma-separated list of values or by repeating the
+parameter multiple times with different values as follows:
+
+[,cols="14%,30%,39%,17%",options="header",]
+|=========================================================================
+| Parameter Type | Comma-separated Values | Multiple Parameters | Standard
+| Header | `Header: value1,value2` | `Header: value1, Header: value2`
+| {RFC-7230}#section-3.2.2[RFC 7230 Section 3.2.2]
+
+| Path | `key=value1,value2` | `/key=value1;key=value2`
```
Comment (ePaul) [inline chapters/http-requests.adoc]: I've never found a reason to use a collection path parameter (normally I would use a query parameter instead).

Also, normally path parameters in OpenAPI don't have a key-value syntax – are we introducing something new here?

Comment (tkrop) [reply chapters/http-requests.adoc]: I agree. This is a very rare case. I just added this line for completeness. Should I remove it?

**Comment by maxim-tschumak** [comment]
:+1:

File: chapters/http-requests.adoc (hunk 3 above)
Comment (ePaul) [reply chapters/http-requests.adoc]: Yes, as long as we don't have anywhere else those key-value-parameters in a path, I would omit them here.

**Comment by whiskeysierra** [comment]
:+1:

**Comment by vadeg** [comment]
:

**Comment by zeitlinger** [comment]
:+1:

**Comment by tkrop** [comment]
:+1:

**Comment by zeitlinger** [comment]
:+1:

**Pull request merged**